tests/datasets/test_datasets.py

```diff
@@ -13,363 +13,433 @@
-            datasets.get(remote_file, path)
-
-
-def _validate_local_tree(root, tree):
-    for path in tree:
-        full_path = os.path.join(root, path)
-        if full_path.endswith("/"):
-            assert os.path.isdir(full_path)
-        else:
-            with open(full_path, "rb") as f:
-                content = f.read()
-            assert content == TEST_FILE_CONTENT
-
-
-@pytest.mark.parametrize("path", VALID_ROOT_PATHS)
-def test_get_tree(remote_tree, path):
-    with temporary_directory() as dirname:
-        datasets.get(path, dirname)
-        _validate_local_tree(dirname, TEST_TREE)
-
-
-@pytest.mark.parametrize("path", VALID_ROOT_PATHS)
-def test_get_tree_in_directory(remote_tree, path):
-    with temporary_directory() as dirname:
-        # Ensure dirname ends with exactly one '/'
-        dirname = dirname.rstrip("/") + "/"
-        datasets.get(path, dirname)
-        _validate_local_tree(dirname, TEST_TREE)
-
-
-@pytest.mark.parametrize("path", VALID_DIRECTORIES)
-def test_get_subtree(remote_tree, path):
-    prefix = path.strip("/") + "/"
-    subtree = [
-        tree_path[len(prefix) :]
-        for tree_path in TEST_TREE
-        if tree_path.startswith(prefix)
-    ]
-    with temporary_directory() as dirname:
-        datasets.get(path, dirname)
-        _validate_local_tree(dirname, subtree)
-
-
-@pytest.mark.parametrize("path", VALID_DIRECTORIES)
-def test_get_subtree_in_directory(remote_tree, path):
-    prefix = path.strip("/") + "/"
-    subtree = [
-        tree_path[len(prefix) :]
-        for tree_path in TEST_TREE
-        if tree_path.startswith(prefix)
-    ]
-    with temporary_directory() as dirname:
-        # Ensure dirname ends with exactly one '/'
-        dirname = dirname.rstrip("/") + "/"
-        datasets.get(path, dirname)
-        _validate_local_tree(dirname, subtree)
-
-
-@pytest.mark.parametrize("path", VALID_DIRECTORIES)
-def test_get_tree_bad_local_path(remote_tree, path):
-    with temporary_directory() as dirname:
-        local_path = os.path.join(dirname, "directory/does/not/exist/")
-        with pytest.raises(IOError):
-            datasets.get(path, local_path)
+# from tests.datasets.fixtures import (
+#     read_remote_object,
+#     temporary_directory,
+#     TEST_DIRECTORY,
+#     TEST_FILE_NAME,
+#     TEST_TREE,
+#     TEST_TREE_NO_HIDDEN_FILES,
+#     VALID_ROOT_PATHS,
+#     VALID_DIRECTORIES,
+#     VALID_FILES,
+#     INVALID_PATHS,
+#     TEST_FILE_CONTENT,
+#     TEST_NON_EXISTENT,
+#     EMPTY_DIRECTORY,
+# )
+
+
+# pytestmark = pytest.mark.usefixtures("project_directory")
+
+
+@pytest.fixture
+def mock_client(mocker):
+    session = mocker.Mock()
+    get_session_mock = mocker.patch(
+        "faculty.datasets.get_session", return_value=session
+    )
+
+    project_id = mocker.Mock()
+
+    object_client = mocker.Mock()
+    mocker.patch("faculty.datasets.ObjectClient", return_value=object_client)
+
+    yield [object_client, project_id]
```

Comment:
Nit: I think this would make more sense as a tuple:
```suggestion
    yield (object_client, project_id)
```
or even just:
```suggestion
    yield object_client, project_id
```